### PR TITLE
feat: Refresh design, use latest ManyJs

### PR DESCRIPTION
**Pull request by stanleyjones**
## Description

Kaylee is starting to be useful again, so this PR resurrects it so we can use it to diagnose network requests.

- [x] Use latest ManyJS
- [x] Use new shared UI library
- [x] Update to latest React, Webpack, etc. (so we know how to update Alberto and Gwen too)
- [x] Use Lifted logo instead of L1
- [ ] ~Poll network for attribute information, show status~ (future)

## Related Issue

Nothing yet, but relates to the "Allow frontends to connect to arbitrary backends by URL" story that @tysloan is still getting ready.

## Screenshots (if applicable)

**OLD KAYLEE**

Screenshot 2022-11-21 at 3 57 28 PM

**NEW KAYLEE**

Screenshot 2022-11-21 at 3 51 08 PM

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+18.12.1
```
Comment (stanleyjones) [inline .nvmrc:1]: Use current Node LTS.

File: config-overrides.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  config.resolve.fallback = {
+    // url: require.resolve('url'),
+    // fs: require.resolve('fs'),
+    // assert: require.resolve('assert'),
+    // crypto: require.resolve('crypto-browserify'),
+    // http: require.resolve('stream-http'),
+    // https: require.resolve('https-browserify'),
+    // os: require.resolve('os-browserify/browser'),
+    buffer: require.resolve("buffer"),
+    stream: require.resolve("stream-browserify"),
```
Comment (stanleyjones) [inline config-overrides.js:13]: The latest Webpack (5+) no longer includes Node polyfills by default, so we have to add them here. We'll need to do this in Alberto and Gwen eventually.

File: config-overrides.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  };
+  config.plugins.push(
+    new webpack.ProvidePlugin({
+      process: "process/browser",
+      Buffer: ["buffer", "Buffer"],
```
Comment (stanleyjones) [inline config-overrides.js:18]: Tell the browser that we mean Node's `Buffer` if it's not defined.

File: package.json [hunk 4]
```diff
@@ -3,26 +3,31 @@
...
-    "@types/react-dom": "^17.0.11",
-    "eslint-config-react-app": "^7.0.0",
-    "many": "l-1-labs/many-js",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "react-scripts": "4.0.3",
-    "typescript": "^4.4.4",
-    "web-vitals": "^1.1.2"
+    "@liftedinit/many-js": "^0.1.0-alpha.5",
+    "@liftedinit/ui": "github:liftedinit/lifted-ui",
```
Comment (stanleyjones) [inline package.json:7]: Use the latest libraries from Lifted.

File: package.json [hunk 5]
```diff
@@ -3,26 +3,31 @@
...
+    "buffer": "^6.0.3",
+    "react": "^18.2.0",
+    "react-app-rewired": "^2.2.1",
```
Comment (stanleyjones) [inline package.json:10]: React App Rewired allows us to modify Webpack (see `config-overrides.js` above) without ejecting from Create React App.

File: src/components/AppProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { ThemeProvider, theme } from "@liftedinit/ui";
+
+interface ProviderProps {
+  children: React.ReactNode;
+}
+
+export default function AppProvider({ children }: ProviderProps) {
+  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
```
Comment (stanleyjones) [inline src/components/AppProvider.tsx:9]: Use the shared UI theme (and all of its styled components).

File: src/components/Identity.tsx [hunk 7]
```diff
@@ -1,63 +1,86 @@
...
-          </form>
-        </Tab>
+    <Box bg="white" p={6}>
+      <Heading>Identity</Heading>
+      <Tabs>
+        <TabList>
+          <Tab onClick={() => setId()}>Anonymous</Tab>
+          <Tab onClick={() => setMnemonic(Id.getMnemonic())}>Random</Tab>
+          <Tab>Seed Words</Tab>
+          <Tab>Import PEM</Tab>
```
Comment (stanleyjones) [inline src/components/Identity.tsx:32]: Currently you can generate an identity from these four sources. Once the "Account Selector" dropdown is properly abstracted from Alberto, we can use that instead. Right now it's connected to all sorts of app state.

File: src/components/Network.tsx [hunk 8]
```diff
@@ -1,28 +1,41 @@
...
-        <label>
-          URL
-          <input name="url" defaultValue="http://localhost:8000" />
-        </label>
-        <br />
-        <button>Connect</button>
-      </form>
-    </Section>
+    <Box bg="white" p={6}>
+      <Heading>Network</Heading>
```
Comment (stanleyjones) [inline src/components/Network.tsx:21]: Same thing for networks. We can eventually use the dropdown here.

File: src/components/Request.tsx [hunk 9]
```diff
@@ -1,63 +1,82 @@
...
+    msg && convertMsg(msg);
+  }, [id, msg]);
+
+  React.useEffect(() => {
+    function hexToJson(hex: string) {
+      const cose = Msg.fromCborData(Buffer.from(hex, "hex"));
+      setJson(cose.toString());
+    }
+    hex && hexToJson(hex);
+  }, [hex]);
```
Comment (stanleyjones) [inline src/components/Request.tsx:50]: You can generate a message from the form _or_ paste hex in directly. Either way will be parsed into the JSON view and set the pending request.

File: src/components/Response.tsx [hunk 10]
```diff
@@ -1,50 +1,42 @@
...
-        <Tab>
-          <pre>{message}</pre>
-        </Tab>
+        <TabPanels>
+          <TabPanel>
+            <Textarea isReadOnly name="hex" value={res} />
+          </TabPanel>
+          <TabPanel>
+            <pre style={{ whiteSpace: "pre-wrap" }}>
+              {res && Msg.fromCborData(Buffer.from(res, "hex")).toString()}
```
Comment (stanleyjones) [inline src/components/Response.tsx:34]: Honestly, I don't like how we're parsing the response into JSON here. I want to do something that's fully recursive so anything that looks like CBOR gets processed and replaced. But this works for now.

File: src/index.tsx [hunk 11]
```diff
@@ -1,10 +1,14 @@
 import React from "react";
-import ReactDOM from "react-dom";
+import { createRoot } from "react-dom/client";
```
Comment (stanleyjones) [inline src/index.tsx:2]: Now React 18 compliant.

File: tsconfig.json [hunk 12]
```diff
@@ -1,26 +1,20 @@
...
     "moduleResolution": "node",
-    "resolveJsonModule": true,
-    "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react-jsx"
+    "noFallthroughCasesInSwitch": true,
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "target": "es2015"
```
Comment (stanleyjones) [inline tsconfig.json:17]: The only change in this file is targeting `es2015` instead of `es5`. The rest is sorting.

**Review by mjsorribas — APPROVED** [review]

**Pull request merged**